core/src/main/java/greencity/service/impl/SocialNetworkImageServiceImpl.java

```diff
@@ -106,6 +111,86 @@ public SocialNetworkImage getSocialNetworkImageByUrl(String url) {
         );
     }
 
+    /**
+     * Method for deleting {@link SocialNetworkImage} by its id.
+     *
+     * @param id - {@link SocialNetworkImage} instance id which will be deleted.
+     */
+    @Override
+    public void delete(Long id) {
+        socialNetworkImageRepo.deleteById(id);
+    }
+
+    /**
+     * Method for deleting all {@link SocialNetworkImage} instance by list of IDs.
+     *
+     * @param listId list of id {@link SocialNetworkImage}
+     */
+    @Override
+    public void deleteAll(List<Long> listId) {
+        listId.forEach(socialNetworkImageRepo::deleteById);
+    }
+
+    @Override
+    public SocialNetworkImageResponseDTO save(SocialNetworkImageRequestDTO socialNetworkImageRequestDTO,
+                                              MultipartFile image) {
+        SocialNetworkImage toSave = modelMapper.map(socialNetworkImageRequestDTO, SocialNetworkImage.class);
+        if (image != null) {
+            toSave.setImagePath(fileService.upload(image).toString());
+        }
+
+        try {
+            socialNetworkImageRepo.save(toSave);
+        } catch (DataIntegrityViolationException e) {
+            throw new NotSavedException(ErrorMessage.SOCIAL_NETWORK_IMAGE_NOT_SAVED);
+        }
+
+        return modelMapper.map(toSave, SocialNetworkImageResponseDTO.class);
+    }
+
+    /**
+     * {@inheritDoc}
+     * Method for finding {@link SocialNetworkImage} by id
+     *
+     * @param id {@link SocialNetworkImage} instance id.
+     * @return {@link SocialNetworkImage}
+     */
+    @Override
+    public SocialNetworkImage findById(Long id) {
```

Comment:
Could you please hide this method by making it private and removing from the interface. 

This way you will not expose `SocialNetworkImage` to the Controller layer. This is what we are trying to achieve in this epic about 3-layers application.